qcodes/instrument/parameter.py

```diff
@@ -900,6 +904,9 @@ class Parameter(_BaseParameter):
        :meth:`set_raw` methods are automatically wrapped to provide ``get`` and
        ``set``.
 
+    To detect if a parameter is gettable or settable check the attributes ``gettable``
+    and ``settable`` on the parameter.
```

Comment:
do we have `hasattr(..)` calls in qcodes? perhaps we should convert those to use these attributes already?

Reply:
Converted the use internally in parameter to use this. This breaks the S46 driver due to https://github.com/QCoDeS/Qcodes/pull/1957/files that monkey patches the get and the set functions.